sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +354,144 @@ export class AssessmentSequentialProcessingService {
+        application: {
+          student: { sinValidation: true, user: true },
+          programYear: true,
+        },
+      },
+      where: {
+        id: assessmentId,
+      },
+    });
+    if (!assessment) {
+      throw new Error(`Assessment if ${assessmentId} not found.`);
+    }
+    const lastName = assessment.application.student.user.lastName;
+    const sin = assessment.application.student.sinValidation.sin;
+    const birthDate = assessment.application.student.birthDate;
+    const programYearStartDate = assessment.application.programYear.startDate;
+    // Get the first assessment date for the current application. If there are multiple
+    // assessments for the current application, then set the first assessment date to
+    // the assessment date of the first assessment.
+    let firstAssessmentDate = getISODateOnlyString(assessment.assessmentDate);
+    const firstAssessment = await this.dataSource.manager
+      .getRepository(StudentAssessment)
+      .createQueryBuilder("assessment")
+      .select("assessment.assessmentDate")
+      .innerJoin("assessment.application", "application")
+      .where("assessment.id = :assessmentId", { assessmentId: assessment.id })
+      .where("application.applicationNumber = :applicationNumber", {
+        applicationNumber: assessment.application.applicationNumber,
+      })
+      .orderBy("assessment.assessmentDate")
+      .getOne();
+    if (firstAssessment) {
+      firstAssessmentDate = getISODateOnlyString(
+        firstAssessment.assessmentDate,
+      );
+    }
+    const sfasApplicationLegacyRepo =
+      sfasApplicationType === "sfasPTApplication"
+        ? this.sfasPartTimeApplicationsRepo
+        : this.sfasApplicationsRepo;
+    const sfasApplicationLegacyAwards = sfasApplicationLegacyRepo
+      .createQueryBuilder(`${sfasApplicationType}`)
+      .select(`SUM(${sfasApplicationType}.csgp_award)`, "CSGP")
+      .addSelect(`SUM(${sfasApplicationType}.sbsd_award)`, "SBSD")
+      .addSelect(`SUM(${sfasApplicationType}.csgd_award)`, "CSGD")
+      .addSelect(`SUM(${sfasApplicationType}.bcag_award)`, "BCAG");
+    if (sfasApplicationType === "sfasPTApplication")
+      sfasApplicationLegacyAwards.addSelect(
+        `SUM(${sfasApplicationType}.cspt_award)`,
+        "CSPT",
+      );
+    sfasApplicationLegacyAwards
+      .innerJoin(`${sfasApplicationType}.individual`, "sfasStudent")
+      .where("lower(sfasStudent.lastName) = lower(:lastName)", { lastName })
+      .andWhere("sfasStudent.sin = :sin", { sin })
+      .andWhere("sfasStudent.birthDate = :birthDate", { birthDate })
+      .andWhere(
+        new Brackets((qb) => {
+          qb.where(`${sfasApplicationType}.startDate >= :startDate`, {
+            startDate: programYearStartDate,
+          }).andWhere(`${sfasApplicationType}.endDate <= :endDate`, {
+            endDate: firstAssessmentDate,
+          });
+        }),
+      );
+    return sfasApplicationLegacyAwards.getRawOne<SFASApplicationAwards>();
```

Comment:
Why can't we use the existing interface to get awards.?

![image](https://github.com/bcgov/SIMS/assets/54600590/557d3f2e-0032-4370-ba71-1ca4c0572aee)

With that in place, we just need to create one method to merge 2 arrays of AwardTotal 

1. AwardTotal[] (SIMS disbursement)
2. AwardTotal[] (Legacy FT)
3. 2. AwardTotal[] (Legacy PT)